lmp/generate-static-deltas

```diff
@@ -104,18 +104,18 @@ def main(creds_zip_file: str, deltas: List[Delta], factory: str, tok_secret_name
     load_extra_certs()
 
     work = 1  # 1 for the fiopush
-    downloads = []
-    pulls = []
+    downloads = set()  # a set of URLs to ostree repo archives to download from Jobserv
+    pulls = set()  # a set of ostree commit hashes to pull from ostreehub (if no URL to archive)
     for d in deltas:
-        downloads.append(d.to[1]) if d.to[1] else pulls.append(d.to[0])
-        work += 1
+        (e, s) = (d.to[1], downloads) if d.to[1] else (d.to[0], pulls)
+        if e not in s:
+            work += 1
+        s.add(e)
```

Comment:
i think you could put `s.add(e)` in the "if" block for a negligible performance improvement.

Reply:
I did so in the first place and then thought that you would ask to move it out of `if` since that `if` is not needed for "unification" of the input list, just for the "work" counting. So, if the work counting is moved or removed then algorithm would still work.